src/xo-app/vm/tab-network.js

```diff
@@ -149,159 +145,226 @@ class VifItem extends BaseComponent {
+                  onChange={newIp => this._saveIp(ipIndex, newIp)}
+                  predicate={this._getIpPredicate()}
+                  resourceSetId={resourceSet}
+                  value={ip}
+                  xoType={resourceSet ? 'resourceSetIp' : 'ip'}
+                >
+                  {ip}
+                </XoSelect>
+              </Col>
+              <Col size={1}>
+                <ActionRowButton
+                  handler={this._deleteIp}
+                  handlerParam={ipIndex}
+                  icon='delete'
+                />
+              </Col>
+            </Row>
+          ))
+        )}
+        <Row>
+          <Col size={10}>
+            {showNewIpForm ? (
+              <span onBlur={this._toggleNewIp}>
+                {resourceSet ? (
+                  <SelectResourceSetIp
+                    autoFocus
+                    containerPredicate={this._getIsNetworkAllowed()}
+                    onChange={ip => this._addIp(ip)}
+                    predicate={this._getIpPredicate()}
+                    required
+                    resourceSetId={resourceSet}
+                  />
                 ) : (
-                  <ActionButton
-                    btnStyle='success'
-                    size='small'
-                    handler={this._toggleNewIp}
-                    icon='add'
+                  <SelectIp
+                    autoFocus
+                    containerPredicate={this._getIsNetworkAllowed()}
+                    onChange={ip => this._addIp(ip)}
+                    predicate={this._getIpPredicate()}
+                    required
                   />
                 )}
-              </Col>
-            </Row>
-          </Container>
-        </td>
-        <td>
+              </span>
+            ) : (
+              <ActionButton
+                btnStyle='success'
+                size='small'
+                handler={this._toggleNewIp}
+                icon='add'
+              />
+            )}
+          </Col>
+        </Row>
+      </Container>
+    )
+  }
+}
+
+const COLUMNS = [
+  {
+    itemRenderer: vif => `VIF #${vif.device}`,
+    name: _('vifDeviceLabel'),
+  },
+  {
+    itemRenderer: vif => (
+      <pre>
+        <Text value={vif.MAC} onChange={mac => setVif(vif, { mac })} />
+      </pre>
+    ),
+    name: _('vifMacLabel'),
+  },
+  {
+    itemRenderer: vif => vif.MTU,
+    name: _('vifMtuLabel'),
+  },
+  {
+    component: connectStore(() => {
+      const getVif = (state, props) => props.item
+      const getNetworkId = createSelector(getVif, vif => vif.$network)
+      const getNetwork = createGetObject(getNetworkId)
+
+      return (state, props) => ({
+        vif: getVif(state, props),
+        network: getNetwork(state, props),
+        networkId: getNetworkId(state, props),
+      })
+    })(({ vif, network, networkId }) => {
+      const _setNetwork = network => setVif(vif, { network })
+      const _getNetworkPredicate = createSelector(
```

Comment:
You're creating a selector inside of the component's render function and therefore losing the main feature of a selector. A new selector means a new result cache and thus, the callback will be called on each render. If you need selectors that depend on the props, you'll either need to create them in the `connectStore` of create a stateful component to create the selectors inside the class.